rest/tests/test_search.py

```diff
@@ -0,0 +1,303 @@
+"""
+Tests for search endpoint.
+"""
+
+from __future__ import unicode_literals
+
+from six.moves import urllib_parse
+from lxml import etree
+
+from rest_framework.status import (
+    HTTP_200_OK,
+    HTTP_405_METHOD_NOT_ALLOWED,
+)
+
+from learningresources.api import create_repo
+from learningresources.models import LearningResource, get_preview_url
+from rest.tests.base import (
+    RESTTestCase,
+    REPO_BASE,
+    as_json,
+)
+from taxonomy.models import Term, Vocabulary
+
+
+class TestSearch(RESTTestCase):
+    """
+    Tests for search endpoint.
+    """
+
+    def get_results(self, query="", selected_facets=None):
+        """Helper method to get search results."""
+        if selected_facets is None:
+            selected_facets = []
+
+        selected_facets_arg = ""
+        for facet in selected_facets:
+            selected_facets_arg += "&selected_facets={facet}".format(
+                facet=urllib_parse.quote_plus(facet)
+            )
+        resp = self.client.get(
+            "{repo_base}{repo_slug}/search/?q={query}{facets}".format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+                query=urllib_parse.quote_plus(query),
+                facets=selected_facets_arg
+            )
+        )
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        return as_json(resp)
+
+    def assertResultEqual(self, result, resource):
```

Comment:
I think this should be changed, especially to make sure there is no confusion with the `unitest` methods